raster/r.geomorphon/main.c

```diff
@@ -309,6 +455,23 @@ int main(int argc, char **argv)
         G_verbose_message("Flat threshold distance m: %f, height: %f",
                           flat_distance, flat_threshold_height);
         G_verbose_message("%s version", (extended) ? "Extended" : "Basic");
+        /*
+         * FIXME: It would be nice to have an additional "-s" flag to set the
+         * current computational region just large enough for the one-off
+         * computation (plus some margins and any required outward alignment
+         * to the cell boundary). And also perhaps another "-r" flag to
+         * restore the region afterwards.
+         */
+        if (oneoff) {
+            unsigned long window_square = nrows * ncols;
+            unsigned long search_square = 4 * search_cells * search_cells;
+
+            if (window_square > 100000000 &&
+                window_square / search_square > 10)
+                G_warning(_("There may be a notable processing delay because the "
+                           "computational region is %lu times larger than necessary"),
+                          window_square / search_square);
```

Comment:
Just a note that I appreciate this form of check esp. that it is reported only when the region is large. It should fit well with usual workflows in GRASS GIS.

Reply:
The use case magnitude is gigacells, so when the region size overshoots it takes many minutes just to run through the buffer dry.